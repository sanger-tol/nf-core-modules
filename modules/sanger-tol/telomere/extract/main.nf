process TELOMERE_EXTRACT {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gawk:5.3.0' :
        'biocontainers/gawk:5.3.0' }"

    input:
    tuple val(meta), path(telomere)

    output:
    tuple val(meta), path("*.bed")      , emit: bed
    tuple val(meta), path("*.bedgraph") , emit: bedgraph
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix  = task.ext.prefix ?: "${meta.id}"
    """
    awk 'BEGIN { OFS = "\t" }
    {
        gsub(">", "")
        print \$2, \$4, \$5 >> "${prefix}_telomere.bed"
        print \$2, \$4, \$5, (((\$5-\$4)<0)?-(\$5-\$4):(\$5-\$4)) >> "${prefix}_telomere.bedgraph"
    }' ${telomere}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gawk: \$(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
    END_VERSIONS
    """

    stub:
    def prefix  = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_telomere.bed
    touch ${prefix}_telomere.bedgraph

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gawk: \$(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
    END_VERSIONS
    """
}
