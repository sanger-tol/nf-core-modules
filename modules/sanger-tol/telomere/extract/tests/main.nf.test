nextflow_process {

    name "Test Process TELOMERE_EXTRACT"
    script "../main.nf"
    process "TELOMERE_EXTRACT"

    tag "modules"
    tag "modules_sangertol"
    tag "telomere"
    tag "telomere/extract"

    test("Cloeon_dipterum - telomere") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'Cloeon_dipterum' ], // meta map
                    file(params.modules_testdata_base_path + 'resources/modules/telomere/idFanCani4/idFanCani4.windows', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Cloeon_dipterum - telomere - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'Cloeon_dipterum' ], // meta map
                    file(params.modules_testdata_base_path + 'resources/modules/telomere/idFanCani4/idFanCani4.windows', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
