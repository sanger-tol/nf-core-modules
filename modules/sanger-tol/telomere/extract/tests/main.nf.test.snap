{
    "Cloeon_dipterum - telomere": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bed:md5,dcc1474c5421f788ee642a9caa6f8224"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bedgraph:md5,b62d187a03802d145039b88353aa3a2a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d8ed4606dcdc23a881780fae42073223"
                ],
                "bed": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bed:md5,dcc1474c5421f788ee642a9caa6f8224"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bedgraph:md5,b62d187a03802d145039b88353aa3a2a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8ed4606dcdc23a881780fae42073223"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-10T14:42:08.584488667"
    },
    "Cloeon_dipterum - telomere - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d8ed4606dcdc23a881780fae42073223"
                ],
                "bed": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedgraph": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum_telomere.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8ed4606dcdc23a881780fae42073223"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-09T17:17:30.038590765"
    }
}